.github/actions/verify-metrics-snapshot/action.yaml

```diff
@@ -0,0 +1,86 @@
+# Copyright (c) 2023 The Jaeger Authors.
+# SPDX-License-Identifier: Apache-2.0
+
+
+
+name: 'Verify Metric Snapshot and Upload Metrics'
+description: 'Upload or cache the metrics data after verification'
+inputs:
+  snapshot:
+    description: 'Path to the metric file'
+    required: true
+  artifact_key:
+    description: 'Artifact key used for uploading and fetching artifacts'
+    required: true
+  cache_key:
+    description: 'Cache key used for uploading and fetching the correct cached metric'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Upload Metrics Artifact
+      uses: actions/upload-artifact@6f51ac03b9356f520e9adb1b1b7802705f340c2b # v4.5.0
+      with:
+        name: ${{ inputs.artifact_key }}
+        path: ./.metrics/${{ inputs.snapshot }}
+        retention-days: 7
+
+    - name: Get v2 tag 
+      shell: bash
+      run: |
+        v2_tag=$(make echo-v2)
+        echo "v2_tag=$v2_tag" >> $GITHUB_ENV
+        echo "$v2_tag"
+
+    - name: Check existing cache
+      id: check-cache
+      uses: actions/cache/restore@1bd1e32a3bdc45362d1e726936510720a7c30a57
+      with:
+        path: ./.metrics/release_${{ inputs.snapshot}}
+        key: ${{ env.v2_tag }}_${{ inputs.artifact_key }}
+        lookup-only: true
+  
+      # The github cache restore successfully restores when cache saved has same key and same path.
+      # Hence to restore release metric with name relese_{metric_name} , the name must be changed to the same.
+    - name: Change file name before caching
+      if: ${{ (github.ref_name == 'main') && (steps.check-cache.outputs.cache-hit != 'true') }}
```

Comment:
I now understand why you're changing the name, but I don't understand why it needs to depend on a cache lookup? Why wouldn't we always want to upload/cache on the main branch? 

Also, we originally talked about caching for tagged release, but now you're doing it for all commits on main? It's actually a good idea because otherwise multiple PRs since a release might be showing cumulative diffs, while we only want the diff vs. main